RELEASES.md

```diff
@@ -96,38 +96,58 @@ deadline of 3 weeks.
 
 ### Cutting a major release
```

Comment:
i kinda want to change this to minor as that is more ped/semantically correct

Reply:
Honestly, we should probably move to date-based release numbering: `202403.0`, `202403.1`, etc.

I don't think there will ever be, on this codebase, a `2.x` release.

Reply:
we could stick with semantic releases and bump minor to major - ie next (major) release would be 31.0

cc @mattklein123 who has suggested something similar in the past i think

Reply:
> Honestly, we should probably move to date-based release numbering: 202403.0, 202403.1, etc.

This seems optimal to me given the realities that @ggreenway mentions. There will never be a 2.x. Not sure how much pain if any this would cause people.

Reply:
yeah, i guess it carries more information, and is strictly speaking semantic, so wont break version parsing

lets tackle this separately - ill open an issue for it now

Reply:
#33910